### ISSUE TITLE
Support for Passing Backstage Token on Request

### ISSUE THREAD

**Issue by Jonas-Beck** [sysdiglabs/backstage-plugin-sysdig#16]
# Description
In Backstage version 1.28.0, the default value of the proxy's `credentials` field was changed to `require` from `dangerously-allow-unauthenticated`, meaning that all requests must include a valid Backstage token.

This update causes the Sysdig plugin requests to return a Status `401 (Unauthorized)` error when using the proxy settings defined in the documentation.

## Current Workaround:

The quick fix for this issue is to update the proxy configuration with `credentials: dangerously-allow-unauthenticated`. This solves the problem but also opens up the proxy to unauthenticated users.

```diff
proxy:
  endpoints:
     '/sysdig':
       target: ${SYSDIG_SECURE_ENDPOINT}
       changeOrigin: true
       allowedMethods: ['GET']
+      credentials: dangerously-allow-unauthenticated
       headers:
         "Authorization": "Bearer ${SYSDIG_SECURE_TOKEN}"
         "Content-Type": "application/json"
         "Accept": "application/json"
         "X-Sysdig-Product": "SDS"
```

## Proposed Solution:

A better solution would be to add support for the plugin to include a valid Backstage token with each request. This can be done by using the Backstage `FetchApi.fetch()` method instead of the standard `fetch()`.

## References:
- [Backstage FetchApi](https://backstage.io/docs/reference/core-plugin-api.fetchapi/)
- [Backstage Version 1.28.0 Release Notes](https://backstage.io/docs/releases/v1.28.0)

**Comment by mbaylis-coles** [comment]
After a day of troubleshooting this workaround fixed my issue

**Comment by AhmedRaafat14** [comment]
Hi, Thanks for the MR fixing this.

Just tested it (v1.3.2) and it works fine 🚀

**Comment by Jonas-Beck** [comment]
This issue has been resolved in version v1.3.2. Closing the issue now.
